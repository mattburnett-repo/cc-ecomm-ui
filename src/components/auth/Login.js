export default function Login () {
    return (
        <div>
            <div>
                <button aria-label="google-auth">Log in with Google</button>
                <button aria-label="github-auth">Log in with Google</button>
            </div>
            <div>
                <label htmlFor="username">Username:</label>
                <input id="username" aria-label="username" />
            </div>
            <div>
                <label htmlFor="password">Password:</label>
                <input id="password" type="password" aria-label="password" />
            </div>
            <div>
                <button aria-label="login">Log in</button>
            </div>
            <div>
            <label htmlFor="register">Don't have an account yet? </label>
                <a href="/register" id="register" aria-label="register">Register</a>
            </div>
        </div>
    )
}