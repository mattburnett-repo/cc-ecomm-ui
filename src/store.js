import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;