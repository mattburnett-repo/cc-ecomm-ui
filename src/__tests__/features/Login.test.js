import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';
import store from '../../store.js'

import Login from '../../features/auth/login';

describe('<Login /> unit / functional tests', () => { 
    beforeEach(() => render(<Provider store={store}><Login /></Provider>)); // need to wrap in Provider because we're using redux in the feature

    it('clicks the Google auth link', () => {       // TODO: mock the actions calls for the login stuff
        const theVal = screen.getByRole('link', {name: /google-auth/i});
        fireEvent.click(theVal);
    });
    it('clicks the GitHub auth button', () => {
        const theVal = screen.getByRole('link', {name: /github-auth/i});
        fireEvent.click(theVal);
    });
    it('accept a value for username', ()=> {
        const theElement = screen.getByRole('textbox', {name: /username/i});
        userEvent.type(theElement, 'asdf');
        
        expect(theElement).toHaveValue('asdf');
    });
    it('accept a value for password', ()=> {
        const theElement = screen.getByLabelText('Password:');
        userEvent.type(theElement, 'asdf');
        
        expect(theElement).toHaveValue('asdf');
    });
    // FIXME: mock the onClick function. right now it triggers the function sends message to console
    it('clicks the Login button and triggers the onSubmit method', () => {
        const theVal = screen.getByRole('button', {name: /login/i});
        fireEvent.click(theVal);
    });
    it('clicks the Register link', () => {
        const theVal = screen.getByRole('link', {name: /register/i});
        fireEvent.click(theVal);
    });
}); // end unit

describe('<Login /> renders a snapshot', () => {
    it('renders a snapshot', () => {
        const tree = renderer.create(<Provider store={store}><Login /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})