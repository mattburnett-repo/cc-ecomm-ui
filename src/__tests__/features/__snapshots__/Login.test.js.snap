// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders a snapshot renders a snapshot 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    aria-label="login-display-form"
    role="presentation"
  >
    <div>
      <a
        aria-label="google-auth"
        href="/auth/google"
        onClick={[Function]}
      >
        Log in with Google
      </a>
    </div>
    <div>
      <a
        aria-label="github-auth"
        href="/auth/github"
        onClick={[Function]}
      >
        Log in with GitHub
      </a>
    </div>
    <div>
      <label
        htmlFor="username"
      >
        Username:
      </label>
      <input
        aria-label="username"
        id="username"
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        aria-label="password"
        id="password"
        type="password"
      />
    </div>
    <div>
      <button
        aria-label="login"
        type="submit"
      >
        Log in
      </button>
    </div>
    <div>
      <label
        htmlFor="register"
      >
        Don't have an account yet? 
      </label>
      <a
        aria-label="register"
        href="/register"
        id="register"
      >
        Register
      </a>
    </div>
  </div>
</form>
`;
